### PR TITLE
HoTT version of Coq-Equations, relying on dev coq-hott package

### PR DESCRIPTION
**Pull request by mattam82**

### CODE REVIEW

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
+opam-version: "2.0"
+authors: [ "Matthieu Sozeau <matthieu.sozeau@inria.fr>" "Cyprien Mangin <cyprien.mangin@m4x.org>" ]
+maintainer: "matthieu.sozeau@inria.fr"
+homepage: "https://mattam82.github.io/Coq-Equations"
+dev-repo: "git+https://github.com/mattam82/Coq-Equations.git"
+bug-reports: "https://github.com/mattam82/Coq-Equations/issues"
+license: "LGPL 2.1"
+build: [
+  ["./configure.sh HoTT"]
```
Comment (palmskog) [inline extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]:
```suggestion
  ["./configure.sh" "HoTT"]
```
This is what makes the builds fail.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  [make "-j%{jobs}%"]
+]
+install: [
+  [make "install"]
+]
+remove: ["rm" "-R" "%{lib}%/coq/user-contrib/Equations"]
```
Comment (palmskog) [inline extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]:
```suggestion
```
This line should be deleted.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam [hunk 3]
```diff
@@ -0,0 +1,31 @@
+opam-version: "2.0"
+authors: [ "Matthieu Sozeau <matthieu.sozeau@inria.fr>" "Cyprien Mangin <cyprien.mangin@m4x.org>" ]
+maintainer: "matthieu.sozeau@inria.fr"
+homepage: "https://mattam82.github.io/Coq-Equations"
+dev-repo: "git+https://github.com/mattam82/Coq-Equations.git"
+bug-reports: "https://github.com/mattam82/Coq-Equations/issues"
+license: "LGPL 2.1"
```
Comment (palmskog) [inline extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]:
```suggestion
license: "LGPL-2.1-or-later"
```
Is this "LGPL 2.1 or later" or "LGPL 2.1 only"? The suggestion is the SPDX identifier for the first case.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  "coq-hott" {= "dev" }
+]
+description: """
+Equations is a function definition plugin for Coq, that allows the
+definition of functions by dependent pattern-matching and well-founded,
+mutual or nested structural recursion and compiles them into core
+terms. It automatically derives the clauses equations, the graph of the
+function and its associated elimination principle.
+"""
+synopsis: "A function definition package for Coq+HoTT"
```
Comment (palmskog) [inline extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]:
```suggestion
synopsis: "A function definition package for Coq+HoTT"
tags: [
  "keyword:dependent pattern-matching"
  "keyword:functional elimination"
  "category:Miscellaneous/Coq Extensions"
  "logpath:Equations"
]
```
Suggested metadata from the regular equations package.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  [make "-j%{jobs}%"]
+]
+install: [
+  [make "install"]
+]
+remove: ["rm" "-R" "%{lib}%/coq/user-contrib/Equations"]
+depends: [
+  "ocaml"
+  "coq" {= "dev" }
+  "coq-hott" {= "dev" }
```
Comment (palmskog) [inline extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]: Are you sure this should be a "version" of the `coq-equations` package as opposed to a package of its own (say, `coq-equations-hott`)? Note that as a version, it won't be possible to both have the regular `coq-equations.dev` version and the `coq-equations-dev+HoTT` version installed at the same time.

**Comment by mattam82** [comment]
Thanks for the review, I’ll fix that. About the version, I hesitated, but indeed it’s not possible to have both versions installed for the same Coq right now (they would both go in user-contribs/Equations). So I decided it was not worth another package at this time. Might revise this later.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam (hunk 3 above)
Comment (mattam82) [reply extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]: Or later was intended indeed.

File: extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam (hunk 4 above)
Comment (mattam82) [reply extra-dev/packages/coq-equations/coq-equations.dev+HoTT/opam]: Thanks

**Comment by palmskog** [comment]
Thanks, this fixes the build issue, but for some reason the `4.05.0` CI job fails when installing the `dev` version of the Coq OPAM package:
```
#=== ERROR while compiling coq.dev ============================================#
...
# Error: failed to run ocaml
```
I reran the job once, but got the same result. Should we escalate this somewhere? I'd be fine with merging the PR in any case.

**Comment by mattam82** [comment]
That looks quite independent indeed.

**Comment by palmskog** [comment]
OK, let's just merge this and open an issue on the archive later.

**Pull request merged**